android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/StringExtensions.kt

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2021 Ona Systems, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartregister.fhircore.engine.util.extension
+
+import java.util.LinkedList
+
+/**
+ * This function replaces the content enclosed within [substitutionPair] with the value obtained
+ * from the [valuesMap]. If the map does not contain the key, the key is returned instead. Default
+ * template is @{key}
+ *
+ * Examples: Given the string "HIV status: @{hivResult}" with a map of {'hivResult': "+ve"}, the
+ * resulting string will be:
+ *
+ * "HIV status: +ve"
+ *
+ * If the key is not available in the map the resulting string will be:
+ *
+ * "HIV status: @{hivResult}"
+ */
+fun String.interpolate(
```

Comment:
This looks sufficient for roll your own iteration for this release, so I wouldn't want to add time or effort to this for now, but did you consider using an existing templating engine? If that's been thought through let's discuss, if not I'd like us to consider using,for example, something like [Velocity](https://velocity.apache.org/engine/devel/developer-guide.html#how-velocity-works)

Reply:
We actually discussed this with @ndegwamartin on Friday.

Reply:
MLS uses a utility class from apache common library.

Reply:
We can refactor the code to use the same. https://github.com/opensrp/fhircore/pull/1455/files

Reply:
Cool, can you create an issue for that? We don't need to worry about it before merge, we can revisit the option of another library if we want to do anything more complex (like loops) in the templating